### PR TITLE
fix: further ease the restriction of executing SQLs in new GRPC interface

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Follow up of #794 .

Carry not recordbatch result in FlightData, to allow executing SQLs other than selection in new GRPC interface.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/datanode/src/instance/flight/stream.rs [hunk 1]
```diff
@@ -31,14 +31,14 @@ use crate::error;
 use crate::instance::flight::TonicResult;
 
 #[pin_project(PinnedDrop)]
-pub(super) struct GetStream {
+pub(super) struct FlightRecordbatchStream {
```
Comment (fengjiachun) [inline src/datanode/src/instance/flight/stream.rs:34]:
```suggestion
pub(super) struct FlightRecordBatchStream {
```

File: src/datanode/src/instance/flight/stream.rs [hunk 2]
```diff
@@ -139,7 +139,7 @@ mod test {
     use super::*;
 
     #[tokio::test]
-    async fn test_get_stream() {
+    async fn test_flight_recordbatch_stream() {
```
Comment (fengjiachun) [inline src/datanode/src/instance/flight/stream.rs:142]:
```suggestion
    async fn test_flight_record_batch_stream() {
```

**Review by fengjiachun — APPROVED** [review]
lgtm

File: src/api/greptime/v1/database.proto [hunk 3]
```diff
@@ -56,3 +56,7 @@ message ObjectResult {
 message FlightDataRaw {
   repeated bytes raw_data = 1;
 }
+
+message FlightAppMeta {
+  uint32 affected_rows = 1;
+}
```
Comment (waynexia) [inline src/api/greptime/v1/database.proto:62]: Maybe it's better to put `affected_rows` as part of data? IMO metadata is something more generic like our `Header`. And in `Output` it is at the same level as `RecordBatch`. To my understanding, `RecordBatch` is for query request and `affected_rows` is for other DML (except `SELECT`) and DDL.

**Comment by MichaelScofield** [comment]
@waynexia PTAL

**Review by waynexia — APPROVED** [review]
I haven't looked into the details, but it generally looks good.

**Pull request merged**